src/net.h

```diff
@@ -253,42 +256,92 @@ class CNetMessage {
     }
 };
 
-/** The TransportDeserializer takes care of holding and deserializing the
- * network receive buffer. It can deserialize the network buffer into a
- * transport protocol agnostic CNetMessage (message type & payload)
- */
-class TransportDeserializer {
+/** The Transport deserializes received messages, and serializes sent messages. */
+class Transport {
 public:
+    virtual ~Transport() {}
+
+    // 1. Receiver side functions, for deserializing the network buffer into a transport protocol
+    // agnostic CNetMessage (message type & payload).
+
     // returns true if the current deserialization is complete
-    virtual bool Complete() const = 0;
-    // set the serialization context version
-    virtual void SetVersion(int version) = 0;
-    /** read and deserialize data, advances msg_bytes data pointer */
-    virtual int Read(Span<const uint8_t>& msg_bytes) = 0;
+    virtual bool ReceivedMessageComplete() const = 0;
+    // set the deserialization context version
+    virtual void SetReceiveVersion(int version) = 0;
+    /** read and deserialize data, advances msg_bytes data pointer. Return false is invalid. */
+    virtual bool ReceivedBytes(Span<const uint8_t>& msg_bytes) = 0;
```

Comment:
`s/false is invalid/false if invalid/` or even better, tell doxygen about it:

```suggestion
    /**
     * Read and deserialize data, advance msg_bytes data pointer.
     * @return false if invalid.
     */
    virtual bool ReceivedBytes(Span<const uint8_t>& msg_bytes) = 0;
```